github/Team.py

```diff
@@ -249,6 +268,26 @@ def set_repo_permission(self, repo, permission):
             "PUT", self.url + "/repos/" + repo._identity, input=put_parameters
         )
 
+    def update_team_repository(self, repo, permission):
+        """
+        :calls: `PUT /orgs/:org/teams/:team_slug/repos/:owner/:repo <https://developer.github.com/v3/teams/#add-or-update-team-repository>`_
+        :param repo: :class:`github.Repository.Repository`
+        :param permission: string
+        :rtype: None
+        """
+
+        # TODO Add support for Content-Length: 0 when no parameters are added
+        # Original note from the docs:
+        # Note that, if you choose not to pass any parameters, you'll need to set Content-Length to zero when calling out to this endpoint. For more information, see "HTTP verbs."
+
+        assert isinstance(repo, github.Repository.Repository), repo
```

Comment:
Why would we call this with no parameters? I don't understand.

Further, you could make this API a little simpler to use to by also accepting the repo name as a string.

Reply:
> Why would we call this with no parameters? I don't understand.

This sentence is a copy of the docs. See https://developer.github.com/v3/teams/#add-or-update-team-repository
In the docs, only the permission parameter is an optional string. If no permission is added, the API will behave like:

> If no permission is specified, the team's permission attribute will be used to determine what permission to grant the team on this repository.

Do you prefer owner + repo in one string (e.g. `PyGithub/PyGithub`) or owner and repo in two parameters (Owner: `PyGithub`, repo: `PyGithub`) or only the repo, because the API assumes that the owner == org?

Reply:
I wasn't going to get so down into the weeds as that, I was just suggesting you change the assert to accept a str as well and passing that in.

Reply:
Added in https://github.com/PyGithub/PyGithub/pull/1509/commits/4b7126cbb6d991967b50ca5ad86e500bc6798e00